### PR TITLE
zadanie 2

### PR DESCRIPTION
**Pull request by KLTPL**

### CODE REVIEW

**Comment by Rafzzz3** [comment]
Zadanie wykonane bardzo dobrze, jeżeli mamy czepiać się detali to:
- w przypadku konwersji warto to robić w bardziej jawyn sposób .astype(int) może być int32 albo int64 w zależności od systemu operacyjnego, warto doprecyzowywać takie rzeczy, ponieważ mało nas to kosztuje a kod jest bardziej przewidywalny 
- warto zauważyć, że XGBoost ma natywne wykrywanie brakujących więc warto by się zastanowić czy opłaca nam się sztucznie wypełniać braki w danych, w przypadku drzew wzmacniających 
- możnaby trochę poprawić jakość komentarzy
Poza w.w uwagami wszystko w porządku

**Comment by Miloproplayer** [comment]
Uważam, że zadanie wykonane jest jak najbardziej poprawnie. 
Jedyne co znalazłem to ustawienie strategy DummyClassifier na uniform, które zakłada równomierny rozkład, co może być mało miarodajne, bo na Titanicu był niezbalansowany rozkład przeżyć do zgonów, . Lepsze zastosowanie to np. 'most-frequent'. Też jest mała nieścisłość we wnioskach, bo dokładność około 85 procent model uzyskał na zbiorze treningowym, a na testowym około 80.

**Comment by jgilewicz** [comment]
Zadanie wykonane prawidłowo, wszystko dokładnie i czytelnie opisane. Z rzeczy na przyszłość:
- warto dla grid searchu wrzucić kilka metryk niż jedna i wykorzystać weighted accuracy, ponieważ dane w zbiorze są niezbalansowane (accuracy dla dummy ok. 60 % > 50%), przykładowo: 
```
gs = GridSearchCV(
    model,
    param_grid,
    scoring=["f1", "average_precision", "accuracy"],
    refit="average_precision",
    cv=5
)
```
-  _nieco większy wynik na zbiorze treningowym świadczy że nie doszło do przetrenowania_ - błędny wniosek, overfitting charakteryzuje się wysokim wynikiem na zbiorze treningowym i super niskim na testowym 🤓 
- fajne wykresy hiperparametrów +

Zadanie zaliczone 👍🏻
